go-controller/pkg/ovn/controller/egressservice/egressservice_zone.go

```diff
@@ -451,51 +463,60 @@ func (c *Controller) repair() error {
 	}
 
 	if config.OVNKubernetesFeature.EnableInterconnect {
-		lrsrPredicate := func(item *nbdb.LogicalRouterStaticRoute) bool {
+		lrpICPredicate := func(item *nbdb.LogicalRouterPolicy) bool {
+			if item.Priority != ovntypes.EgressSVCReroutePriority {
+				return false
+			}
 			svcKey, found := item.ExternalIDs[svcExternalIDKey]
 			if !found {
 				return false
 			}
 
-			if item.Policy == nil || *item.Policy != nbdb.LogicalRouterStaticRoutePolicySrcIP {
-				klog.Infof("Egress service repair will delete lrsr for service %s it has an invalid policy: %v", svcKey, item)
-				return true
-			}
-
 			svc, found := c.services[svcKey]
+			if found {
+				klog.Infof("Egress service repair already finished repairing lrp for service %s: %v", svcKey, item)
+				return false
+			}
 			if !found {
-				klog.Infof("Egress service repair will delete lrsr for service %s because it is no longer a valid egress service: %v", svcKey, item)
-				return true
+				trimmedKeyValue := strings.TrimSuffix(svcKey, interconnectSuffix)
```

Comment:
I know it's far fetched, but is this suffix something user could potentially add to the name that we are using as key and confuse us into thinking we added it?

Reply:
I guess it wouldn't matter if we always check it out first without trimming the suffix, looks like risky business though going forward maintainability-wise if that's the case

Reply:
its a tech preview feature :) it can be fixed moving forward? doesn't even need to be upgrades compatible until this GA's I guess

Reply:
Okay, but perhaps it is just as easy as replacing the `-` in `-ic` for `:` like `:ic`. I think `:` in the CRD name would not be valid.

Reply:
I thought the same thing @jcaamano , but as long as the first check we do is for the entire string, and then fallback check to removing -ic check it should be OK?

Reply:
if adding `:ic` makes it more bullet proof I can update that, but want to make sure rest of the PR is ok? I will address this with any other remaining comments.

Reply:
all dashes changes to colon's in main code and UT's. resolving this comment.